### PR TITLE
Add docker build trigger after bundle builds

### LINKED ISSUE #908: Create Docker images off of bundle build
In order to make validating 1.2.0 easier, provide docker images that can be pulled for testing

- [ ] Add new docker references in the staging docker
- [ ] Update the build process to publish the image
  - [dockerfile](https://github.com/opensearch-project/opensearch-build/blob/main/docker/release/dockerfiles/opensearch.al2.dockerfile), https://hub.docker.com/r/opensearchstaging/opensearch
  - [dockerfile](https://github.com/opensearch-project/opensearch-build/blob/main/docker/release/dockerfiles/opensearch-dashboards.al2.dockerfile),  https://hub.docker.com/r/opensearchstaging/opensearch-dashboards

- [ ] Update the notification to include a notice about the docker image

This should allow anyone to create a security/no-security build for OpenSearch and OpenSearch Dashboards

### PR DESCRIPTION
**Pull request by peterzhuamazon**
Signed-off-by: Peter Zhu <zhujiaxi@amazon.com>

### Description
Add docker build trigger after bundle builds
 
### Issues Resolved
#908
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: jenkins/opensearch/Jenkinsfile [hunk 1]
```diff
@@ -134,8 +131,22 @@ pipeline {
             post() {
                 success {
                     node('Jenkins-Agent-al2-x64-c54xlarge-Docker-Host') {
-                        messages = getJenkinsMessages(['build-x64', 'build-arm64'])
-                        publishNotification(":white_check_mark:", "Successful Build", "\n${messages}")
+                        publishNotification(":white_check_mark:", "Successful Build", "\n${getAllJenkinsMessages()}")
+                        script {
+                            def VERSION = sh(script: 'echo ${INPUT_MANIFEST} | grep -Po "[0-9.]+?(?=.yml)"', returnStdout: true).trim()
+                            println VERSION
```
Comment (zelinh) [inline jenkins/opensearch/Jenkinsfile:137]: Can we delete this line since it's used for our testing.

Comment (peterzhuamazon) [reply jenkins/opensearch/Jenkinsfile:137]: I am ok to leave it there as it shows what version we are retrieving.

Comment (peterzhuamazon) [reply jenkins/opensearch/Jenkinsfile]: Change to version + build number for more information overall.

**Review by zelinh — APPROVED** [review]

**Review by peternied — APPROVED** [review]

**Review by peternied — APPROVED** [review]

File: jenkins/opensearch/Jenkinsfile [hunk 2]
```diff
@@ -132,6 +132,21 @@ pipeline {
                 success {
```
Comment (dblock) [inline jenkins/opensearch/Jenkinsfile:132]: I think this should have been a pipeline step as opposed to something that executes upon a successful build. We are also publishing a success notification prematurely, the docker build can fail, but we wouldn't know in Slack.

Comment (peterzhuamazon) [reply jenkins/opensearch/Jenkinsfile:132]: If needed we can block the bundle build completion until docker build completes and return status.

**Pull request merged**